server/service/src/invoice/prescription/update/mod.rs

```diff
@@ -118,7 +118,11 @@ pub fn create_reverse_prescription(
     new_invoice.id = uuid();
     new_invoice.linked_invoice_id = Some(orig_invoice.id.clone());
     new_invoice.is_cancellation = true;
-    new_invoice.verified_datetime = Some(Utc::now().naive_utc());
+    new_invoice.created_datetime = orig_invoice.created_datetime + Duration::seconds(10);
+    new_invoice.picked_datetime = orig_invoice
+        .picked_datetime
+        .map(|dt| dt + Duration::seconds(10));
+    new_invoice.verified_datetime = Some(Utc::now().naive_utc() + Duration::seconds(10));
```

Comment:
Staggering times so that the adjustment stock is calculated after...

Though, I still find it confusing that the prescription that reduces stock is 'Cancelled' and the one that adjusts stock back is 'Verified'... Should be keeping the original as 'Verified' and then creating a 'Cancelled' one to adjust stock back....

Reply:
Yup that's the suggestion... it's confusing for cancelled to be negative and verified to be positive 👀